agent/engine/docker_container_engine.go

```diff
@@ -168,7 +168,27 @@ var scratchCreateLock sync.Mutex
 
 // NewDockerGoClient creates a new DockerGoClient
 func NewDockerGoClient(clientFactory dockerclient.Factory, cfg *config.Config) (DockerClient, error) {
-	client, err := clientFactory.GetDefaultClient()
```

Comment:
I do not understand why this breaks the abstraction since GetDefaultClient is already in the Factory interface.  Could you clarify what you mean here?